### DISCUSSION TITLE
Sim800 GSM driver

### DISCUSSION THREAD

**Discussion by YogPanjarale** [toitlang/toit#523] [Drivers]
Hello Toit community 👋,
its a awesome platform, I am trying an attempt make a sim800 driver for toit
I Have this [ttgo-tcall-sim800](https://github.com/Xinyuan-LilyGO/LilyGo-T-Call-SIM800) board so I thought can i make it work on toit
I just have cloned [this repo](https://github.com/toitware/sequans-cellular)
and did some changes in it [here's  my repo](https://github.com/YogPanjarale/sim-cellular)
further i'll be changes the at commands and few more changes and try to make it work

**Comment by kasperl** [comment]
Hi Yog!

Welcome to the community :wave: 

Do reach out if you need any help to unblock your progress. A lot of us hang out on Discord!

**Comment by YogPanjarale** [comment]
Hey @kasperl ,
I had a question like how did you start writing the drivers for sequans , ublox , quectel drivers, like 
did you start building upon this or how https://github.com/toitware/cellular/blob/main/src/base.toit
how should i proceed

**Reply by lask** [reply]
Hi @YogPanjarale,
Our experience has been that cellular modems largely follow the same pattern. We have tried to capture that in the `CellularBase` class, so `CellularBase` is a great starting point. The modems are of course not entirely the same, so we overwrite the methods in `CellularBase` that are not the same. The `Cellular` interface defines what a cellular driver should be able to do. I think it is a good idea to go through the methods one by one and implement them. In order to find out what AT commands to send, I find application notes that describe the feature I want. Usually it is sufficient to just send the same AT commands.

The three existing drivers bail out if things go wrong. The idea is that if things don't go as expected, then it is easier to start over and ensure that we are in a state that we like. In other words, it is fine for the driver to fail as long as you implement something that makes it retry the connection until it succeeds.

I hope this can help get you started.
